core/typedef.go

```diff
@@ -526,6 +526,11 @@ func (typeDef *TypeDef) WithObjectConstructor(fn *Function) (*TypeDef, error) {
 	typeDef = typeDef.Clone()
 	fn = fn.Clone()
 	fn.ParentOriginalName = typeDef.AsObject.Value.OriginalName
+	// Constructors are invoked by setting the ObjectName to the name of the object its constructing and the
+	// FunctionName to "", so ignore the name of the function.
+	// This is to be aligned with moduleSchema.typeDefWithObjectConstructor
```

Comment:
Makes sense! We can remove the logic from `moduleSchema.typeDefWithObjectConstructor` then right?